httpx/client.py

```diff
@@ -121,17 +137,17 @@ def merge_url(self, url: URLTypes) -> URL:
     def merge_cookies(
         self, cookies: CookieTypes = None
     ) -> typing.Optional[CookieTypes]:
-        if cookies or self.cookies:
-            merged_cookies = Cookies(self.cookies)
+        if cookies or self._cookies:
```

Comment:
On reflection I think we should just use the public `self.cookies`/`self.headers` accessors for these cases.